src/auto_scheduler/compute_dag.cc

```diff
@@ -665,9 +666,349 @@ ComputeDAG::ComputeDAG(Array<te::Tensor> tensors) {
+                           const Stage& stage, const te::Operation& op,
+                           const te::Tensor& placeholder,
+                           const std::set<std::string>& placeholder_axis_names) {
+  std::ostringstream os;
+  Array<Iterator> stage_iters;
+
+  auto attach_it = state->attach_map->stage_to_attach_iter.find(stage_id);
+  int attach_pos = -1;
+  size_t iters_before_attach = 0;
+  if (attach_it != state->attach_map->stage_to_attach_iter.end()) {
+    auto attach = attach_it->second;
+    const auto& attach_stage = state->stages[attach.first];
+    attach_pos = attach.second;
+    stage_iters.insert(stage_iters.end(), attach_stage->iters.begin(),
+                       attach_stage->iters.begin() + attach_pos + 1);
+  }
+
+  stage_iters.insert(stage_iters.end(), stage->iters.begin(), stage->iters.end());
+
+  std::vector<Iterator> iters;
+  for (size_t i = 0; i < stage_iters.size(); ++i) {
+    const auto& iter = stage_iters[i];
+    if (iter->ori_iters.empty()) {
+      iters.push_back(iter);
+    } else {
+      for (const Iterator& ori_iter : iter->ori_iters) {
+        iters.push_back(ori_iter);
+      }
+    }
+    if (static_cast<int>(i) == attach_pos) {
+      iters_before_attach = iters.size();
+    }
+  }
+
+  std::vector<std::string> new_names;
+  std::vector<std::string> new_axis_names;
+  for (const Iterator& iter : iters) {
+    std::set<std::string> ori_iter_names;
+    ExtractOriginalIterators(iter->name, &ori_iter_names);
+    // fused iters have been replaced with iter->ori_iters.
+    // So there should be only one ori iter name extracted from iter->name.
+    CHECK_EQ(ori_iter_names.size(), 1);
+    auto ori_iter_name = BaseName(*ori_iter_names.begin());
+    new_axis_names.push_back(ori_iter_name);
+  }
+  for (size_t i = 0; i < new_axis_names.size(); ++i) {
+    auto iter = iters[i];
+    std::string ori_iter_name;
+    if (i < iters_before_attach) {
+      ori_iter_name = new_axis_names[i + iters_before_attach];
+    } else {
+      ori_iter_name = new_axis_names[i];
+    }
+    if (placeholder_axis_names.count(ori_iter_name)) {
+      os << iter->range->extent << ori_iter_name;
+      new_names.push_back(ori_iter_name);
+      new_shape->push_back(iter->range->extent);
+    }
+  }
+  std::string new_layout = os.str();
+  os.str("");
+  // TODO(minmin): uncomment this line for relay integration
+  // ::tvm::relay::KernelLayoutTransformer::global_new_layouts_queue.push_back(new_layout);
+  return new_layout;
+}
+
+void ComputeDAG::RewriteLayout(const Array<Step>& transform_steps) {
+  ComputeDAGNode* pdag = this->CopyOnWrite();
+  auto node = make_object<StateNode>();
+  node->transform_steps = transform_steps;
+  node->concrete = true;
+  const State& state = InferBound(State(node));
+  OperationSet handled_ops;
+  int stage_id = -1;
+  for (const auto& stage : state->stages) {
+    stage_id += 1;
+    const te::Operation& op = stage->op;
+    if (op->IsInstance<te::ComputeOpNode>()) {
```

Comment:
Since this statement is pretty long, I'd suggest
```
if (!op->IsInstance<te::ComputeOpNode>()) {
  continue;
}
```
so that we can reduce an indent.